HSPI/Connector.cs

```diff
@@ -11,20 +11,20 @@ public static class Connector
             Justification = "The function wouldn't do anything without a plugin.")]
         [SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes",
             Justification = "I don't know what kinds of exceptions it _could_ throw.")]
-        public static void Connect<TPlugin>(string[] args) where TPlugin : HspiBase, new()
+        public static TPlugin Connect<TPlugin>(string[] args, bool pollForShutdown = true) where TPlugin : HspiBase, new()
         {
```

Comment:
So if pollForShutdown is true, the app will exit when the plugin shuts down, and this method will block until it does. And if it isn’t true, this method will return a plug-in instance. Do multiple plugins get loaded at a time? If one of them is polling for shutdown and shuts down then the whole app will exit. Is this intended?

Reply:
HS _does_ support multiple plugin instances, but that's ultimately going to be different code. The current HSPI code doesn't support multiple instances. Still have to figure out how all that works.